ktor-client/ktor-client-core/common/src/io/ktor/client/features/DefaultRequest.kt

```diff
@@ -37,3 +37,45 @@ public fun HttpClientConfig<*>.defaultRequest(block: HttpRequestBuilder.() -> Un
         block()
     }
 }
+
+/**
+ * Set the [baseUrl] to use it in sub-requests.
+ * To override the existing [baseURL] in a sub-request, the full [URL] is needed in the [HttpRequestBuilder].
```

Comment:
This function is also applicable to a single request so the question is: what will happen if one invokes it outside of call config? The other issue is that it's not clear what is a sub-request for one who doesn't know about the default request feature and out of scope. For sure this should be clarified or somehow fixed.. not sure

Reply:
Yeah, ideally this call should be only available in DefaultRequest feature (hey Multi Receiver Extensions). Until this feature is available, I could subclass the HttpRequestBuilder class and add the baseURL function to the sub-class.

Reply:
I subclassed HttpRequestBuilder and created a DefaultRequest.Builder having the baseURL feature